subprojects/android-test/ui-testing-core-app/src/androidTest/kotlin/com/avito/android/ui/test/ActivityLifecycleStatesTest.kt

```diff
@@ -0,0 +1,52 @@
+package com.avito.android.ui.test
+
+import androidx.lifecycle.Lifecycle
+import com.avito.android.test.app.core.inHouseScreenRule
+import com.avito.android.ui.EmptyActivity
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import ru.avito.util.assertThrows
+
+class ActivityLifecycleStatesTest {
+
+    @get:Rule
+    val activityRule = inHouseScreenRule<EmptyActivity>()
+
+    @Before
+    fun before() {
+        activityRule.launchActivity(null)
```

Comment:
What about cases when activity was not launched?

Reply:
The test will [fail](https://github.com/avito-tech/avito-android/blob/dbdb289b63394d7dcce0a2044445c443dc2830dd/subprojects/test-runner/test-inhouse-runner/src/main/kotlin/com/avito/android/rule/base/BaseActivityScenarioRule.kt#L18) then

Reply:
Ok, should we test it here?

Reply:
It never hurts to check, so added a new test.